Sources/ObjectivelyMVC/Slider.c

```diff
@@ -161,9 +161,13 @@ static _Bool captureEvent(Control *self, const SDL_Event *event) {
 			if (frame.w) {
 
 				const double fraction = (event->motion.x - frame.x) / (double) frame.w;
-				const double value = this->min + (this->max - this->min) * clamp(fraction, 0.0, 1.0);
+				double value = this->min + (this->max - this->min) * clamp(fraction, 0.0, 1.0);
 
-				$(this, setValue, value); // FIXME: Factor step into this
+				if (this->snapToStep) {
+					value = clamp(round(value / this->step) * this->step, this->min, this->max);
```

Comment:
Unless I'm getting overly confused by this, shouldn't it work?
Did some tests in Python to make sure I'm not going insane:
```
>>> min(max(round(5.6 / 0.5) * 0.5, 1.0), 30.0)
5.5
>>> min(max(round(5.8 / 0.5) * 0.5, 1.0), 30.0)
6.0
>>> min(max(round(8.3 / 0.3) * 0.3, 1.0), 30.0)
8.4
```

As far as I can tell, this should work correctly.